src/palace.pyx

```diff
@@ -1701,6 +1703,128 @@ cdef class AuxiliaryEffectSlot:
         """
         return self.impl.get_use_count()
 
+
+cdef class Effect:
+    """
+    A collection of settings or parameters.
+
+    Parameters
+    ----------
+    context : Optional[Context]
+        The context from which the effect is to be created.
+        If it is None, `__init__` does nothing. 
+    """
+    cdef alure.Effect impl
+
+    def __init__(self, context: Optional[Context]) -> None:
+        if context is None: return
+        self.impl = (<Context> context).impl.create_effect()
+
+    def __enter__(self) -> Effect:
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl < other_effect.impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl <= other_effect.impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl == other_effect.impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl != other_effect.impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl > other_effect.impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, Effect):
+            return NotImplemented
+        other_effect: Effect = other
+        return self.impl >= other_effect.impl
```

Comment:
Please see the rest of the current codebase for a less verbose implementation.

Reply:
pls clarify with examples

Reply:
https://github.com/McSinyx/palace/blob/master/src/palace.pyx#L314-L342